### PR TITLE
Passing tests again

### PR DESCRIPTION
**Pull request by pspeter3**
This fixes the tests, specifically for node v0.12. This notably reverts
the fix for the uglify code. I will follow up with a pull request for
that soon. This does mean we can push for #60

### CODE REVIEW

**Comment by pspeter3** [comment]
@evanw and @julien-f This passes tests and improves the fix for #36.

**Comment by julien-f** [comment]
I don't understand why it is not always wanted to remove the headers (as you mentioned [here](https://github.com/evanw/node-source-map-support/issues/36#issuecomment-102679321)), could you explain it in a few world?

Also I think the name of `isFromModule` is not clear enough as it indicates not only if the error originated from a module but also if the error was located on the first line.

**Comment by pspeter3** [comment]
I agree about `isFromModule` is a bad name. The basic summary is that the header is only applied when the stack goes through a native module. I'm not sure exactly why but it is frustrating since it is not consistent. My changes do pass all tests however and solve the uglify case.

**Comment by pspeter3** [comment]
I've updated the code to reflect your changes and hopefully be more extensible. Please merge and publish when convenient

File: test.js [hunk 1]
```diff
@@ -218,7 +218,7 @@ it('function constructor', function() {
     'throw new Function(")");'
   ], [
     'SyntaxError: Unexpected token )',
-    /^    at Object\.Function \((?:unknown source|<anonymous>)\)$/,
+    /^    at Function \((?:unknown source|<anonymous>|native)\)$/,
```
Comment (julien-f) [inline test.js]: I don't understand this change, how is it related to #47?

Comment (julien-f) [inline test.js]: I think this change may be necessary due to some change in Node 0.12 and/or io.js, am I right?
If so, it should be move in its own PR.

Also, simply removing the `Object\.` prefix may be breaking the tests on some platforms.

Comment (pspeter3) [inline test.js]: This is relevant for Node 0.12. I'll move it into its own PR

**Comment by julien-f** [comment]
Fixed in master, thanks.
